gossip/server.go

```diff
@@ -88,6 +88,10 @@ func (s *server) Gossip(argsI proto.Message) (proto.Message, error) {
 	if err != nil {
 		return nil, util.Errorf("local addr %s could not be converted to net.Addr: %s", args.LAddr, err)
 	}
+	// Verify that the client connection is valid and hasn't been closed.
+	if _, ok := s.lAddrMap[lAddr.String()]; !ok {
```

Comment:
I don't understand this change. `lAddr` here is the client's self-reported address, i.e. the address on which it listens, but in the `onOpen`/`onClose` callbacks, we use `conn.RemoteAddr`, which is the ephemeral port used for the other end of this TCP connection, so I don't see how these two could ever match up.